utilities/roswtf/scripts/getlogball.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2016, TORK All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of TORK (Tokyo Opensource Robotics Kyokai Association) 
+#    nor the names of its contributors may be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+FILENAME_LOG_COMMANDS=/tmp/ros_diagnosisinfo_commands_`date +"%Y%m%d-%H%M%S"`.log
+FILENAME_LOG_ROS=/tmp/ros_diagnosisinfo_ros_`date +"%Y%m%d-%H%M%S"`.tgz
+FILENAME_LOG_ALL=/tmp/ros_diagnosisinfo_all_`date +"%Y%m%d-%H%M%S"`.tgz
+
+env | grep -E '(ROS|PATH)' | sort | tee -a ${FILENAME_LOG_COMMANDS}
+ifconfig | tee -a ${FILENAME_LOG_COMMANDS}
+## Get packages' version. Ref. http://askubuntu.com/a/347563/24203
+## Get rosversion of the same packages
+res_dpkg=$(dpkg -l | grep '^ii' | grep ros- | awk '{print $2 " " $3}')
```

Comment:
This is specific to Ubuntu/Debian; what's the rationale for duplicating with the (preferred) logic below based on `rospack list`?

Reply:
While `rosversion` returns what is in `ROS_PACKAGE_PATH` and used by ROS, it's possible the version of a package robot is running on is different from what the users (human) expect. So I thought getting as much info as the installed binary, ROS_PACKAGE_PATH, `rosversion` would be beneficial.
And yes, this particular command is debian/Ubuntu specific, so this can run only on debian/Ubuntu.